zerver/tests/test_subs.py

```diff
@@ -3959,6 +3959,10 @@ def test_guest_user_case() -> None:
 
             self.assertEqual(len(never_subscribed), len(web_public_streams))
 
+            for stream_dict in never_subscribed:
+                self.assertTrue(stream_dict["is_web_public"])
+                self.assertEqual(len(stream_dict["subscribers"]), len(users_to_subscribe))
```

Comment:
The first assertion here belongs in the previous commit, right?  

And an assertion that everything in `never_subscribed` is not in the other lists might also be valuable, since that's the bug the first commit has.